File: 0501. 二叉搜索树中的众数/中序遍历.js
// 92ms
var findMode = function (root) {
    var maxn = 0, result = [], pre = Infinity, count = -1
    function inOrder(root) {
        if (root != null) {
            inOrder(root.left)
            if (root.val == pre) {
                count += 1
            } else {
                if (count > maxn) {
                    maxn = count
                    result = [pre]
                } else if (count == maxn) {
                    result.push(pre)
                }
                pre = root.val
                count = 1
            }
            inOrder(root.right)
        }
    }
    inOrder(root)
    if (count > maxn) {
        maxn = count
        result = [pre]
    } else if (count == maxn) {
        result.push(pre)
    }
    return result
};

File: 0897. 递增顺序搜索树/中序遍历.swift
// 4ms
class Solution {
    func increasingBST(_ root: TreeNode?) -> TreeNode? {
        var nodeList: [TreeNode?] = []
        func inOrder(_ node: TreeNode?) {
            if let node = node {
                inOrder(node.left)
                nodeList.append(node)
                inOrder(node.right)
            }
        }
        inOrder(root)
        nodeList.append(nil)
        for i in 0..<(nodeList.count - 1) {
            nodeList[i]!.left = nil
            nodeList[i]!.right = nodeList[i + 1]
        }
        return nodeList[0]
    }
}